### DISCUSSION TITLE
How to add a commit describing a git filter-repo --path-rename

### DISCUSSION THREAD

**Discussion by gdlxn-ibm** [newren/git-filter-repo#553] [Q&A] [answered]
I would like to rename files and folders in Git, **keeping the commit history** with the new name.  `git filter-repo --path-rename` almost does what I need, **except that I'd like have a commit describing the rename**.

Since `git filter-repo --path-rename`  doesn't add such a commit, I thought of first doing the `git filter-repo --path-rename` and then doing a `git commit` on the  renamed files.  However, Git doesn't seem to (as far I can tell) allow me to stage unchanged files, so I can't really do a `git commit` on the renamed files.

I'm going to next try using a filename callback to see it can be used to add a new commit.  But I thought I'd also ask if anyone knows how to have `git filter-repo --path-rename` add a commit describing the rename?

**Comment by newren** [comment, marked as answer]
I'm afraid I'm not following this description is.  `git filter-repo --path-rename A:B` goes through every commit, and for any that has a file named A will rename it to B.  So, if you have 10,000 commits in your history, and 500 of them make modifications to file A, then --path-rename fixes up 500 commits.  Do you want an additional 500 commits saying that you've renamed A -> B in the history??  Someone reading that would find that really odd.

If you don't want 500 commits but just want one...where do you put that one?  Near the beginning of history?  Near the current end of history?  Why would someone look there to find it?  And what's the purpose of the commit given that after you've rewritten history, `git log` will show that the file has "always" been named B (because that's the point of rewriting history).  So, wouldn't users be confused by this additional commit where you explained why you renamed this file that has always been named B from being named A to being named B?
